src/xarray_regrid/regrid.py

```diff
@@ -125,26 +131,72 @@ def most_common(
 
         Args:
             ds_target_grid: Target grid dataset
-            time_dim: Name of the time dimension. Defaults to "time".
-            max_mem: (Approximate) maximum memory in bytes that the regridding routine
-                can use. Note that this is not the total memory consumption and does not
-                include the size of the final dataset. Defaults to 1e9 (1 GB).
+            expected_groups: Numpy array containing all labels expected to be in the
+                input data. For example, `np.array([0, 2, 4])`, if the data only
+                contains the values 0, 2 and 4.
+            time_dim: Name of the time dimension. Defaults to "time". Use `None` to
+                force regridding over the time dimension.
+            inverse: Find the least-common-value (anti-mode).
 
         Returns:
             Regridded data.
         """
         ds_target_grid = validate_input(self._obj, ds_target_grid, time_dim)
-        return most_common.most_common_wrapper(
-            self._obj, ds_target_grid, time_dim, max_mem
+
+        if isinstance(self._obj, xr.Dataset):
+            msg = (
+                "The 'most common value' regridder is not implemented for\n",
+                "xarray.Dataset, as it requires specifying the expected labels.\n"
+                "Please select only a single variable (as DataArray),\n"
+                " and regrid it separately.",
+            )
+            raise ValueError(msg)
+
+        return flox_reduce.get_most_common_value(
+            self._obj,
+            ds_target_grid,
+            expected_groups,
+            time_dim,
+            inverse,
+        )
+
+    def stat(
+        self,
+        ds_target_grid: xr.Dataset,
+        method: str,
```

Comment:
```suggestion
        method: Literal["sum", "mean", "var", "std", "median"]
```

Reply:
I will still extend it by allow "max" and "min". Any reason why they're not listed here https://flox.readthedocs.io/en/latest/aggregations.html ?

Reply:
looks like i just missed them hehe